### DISCUSSION TITLE
Question about group ID

### DISCUSSION THREAD

**Discussion by Coconut443** [wkentaro/labelme#1138] [Q&A / Help] [answered]
I'm labeling for a instances segmentation project. The number of my class is one. And the class is cow. I notice that there is a new option group ID. I know I need to fill the object label with cow-1, cow-2... But the group ID of them should be the same or not?

**Comment by wkentaro** [comment, marked as answer]
Now you don't need to use cow-X. You can use group ID instead for the "X".
See the latest example: https://github.com/wkentaro/labelme/tree/master/examples/instance_segmentation

In the latest `labelme2voc.py` and `labelme2coco.py`, it assumes that each polygon represents _instance_ except there is no `groupID` specified.

**Comment by Coconut443** [comment]
Thank you!

**Comment by pixx1** [comment]
Does I got it right:
Lets say, I label multiple single cows on an image and I never set any 'group ID'
`labelme2voc.py` and `labelme2coco.py` will not produce instances for each label, but only one?

Am I right, that the json file will always save single polygons/circle/.. as a single instance, no matter if I work with group IDs or not? 

Thanks so far for ur great work and sharing this tool!!

**Comment by wkentaro** [comment]
> labelme2voc.py and labelme2coco.py will not produce instances for each label, but only one?

It WILL produce instances for each label (or each polygon).

> Am I right, that the json file will always save single polygons/circle/.. as a single instance, no matter if I work with group IDs or not?

EACH polygon/circle/.. is saved as a single instance.

**Comment by Joeostr** [comment]
If i wish to classify human faces for gender and age (bounding boxes around the face), then is it a good idea to place the gender class in the label, and the age in the "Group ID", such as the example below (20-30 is the age range) 

![Labelme](https://user-images.githubusercontent.com/31250068/78186936-903c8800-743b-11ea-9127-0d78c72fba24.png)

**Comment by wkentaro** [comment]
I would do Male-2030 in this case.

On Wed, 1 Apr 2020 at 22:10, Joeostr <notifications@github.com> wrote:

> If i wish to classify human faces for gender and age (bounding boxes
> around the face), then is it a good idea to place the gender class in the
> label, and the age in the "Group ID", such as the example below (20-30 is
> the age range)
>
> [image: Labelme]
> <https://user-images.githubusercontent.com/31250068/78186936-903c8800-743b-11ea-9127-0d78c72fba24.png>
>
> —
> You are receiving this because you were assigned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/wkentaro/labelme/issues/633#issuecomment-607491311>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABA4LEYHYNX33KQYXB7RVTLRKOUUHANCNFSM4LUIPV4A>
> .
>
-- 
Kentaro Wada / 和田 健太郎
https://wkentaro.com <http://wkentaro.com>

**Comment by Joeostr** [comment]
You mean not to use the GroupID?

**Comment by wkentaro** [comment]
Yeah. Because Group-Id is not meant to be used for age annotation.

On Thu, 2 Apr 2020 at 17:41, Joeostr <notifications@github.com> wrote:

> You mean not to use the GroupID?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wkentaro/labelme/issues/633#issuecomment-607958655>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABA4LE76AJEBHKA6WGBDB4DRKS52XANCNFSM4LUIPV4A>
> .
>
-- 
Kentaro Wada / 和田 健太郎
https://wkentaro.com <http://wkentaro.com>

**Comment by watertianyi** [comment]
@wkentaro 
How to label a single category with labelme? I read the custom data set, but I didn’t point out a label naming method. Sometimes there is only one category in a picture. Is it cat, cat, cat or cat0, cat1, cat2 or cat-0 , Cat-1, cat-2, thank you!

I use cat-0, cat-1, cat-2 for annotation, and use labelme2coco to convert annotations.json, where the annotation is [], what's the matter?

**Comment by watertianyi** [comment]
> 我
> 说对了吗：可以说，我在一张图像上标记了多头单头母牛，但我从未设置任何“组ID”
> `labelme2voc.py`，`labelme2coco.py`也不会为每个标签生成实例，而只会生成一个实例？
> 
> 我是对的，不管我是否使用组ID，json文件始终将单个多边形/圆形/ ..保存为单个实例？
> 
> 到目前为止，感谢您的出色工作，并分享了此工具！！

How to label a single category with labelme and instance segmentation? There are cat, cat, cat, cat0, cat1, cat2 or cat-0, cat-1, cat-2, etc. which annotation method is used?thank you!

I use cat-0, cat-1, cat-2 for annotation, and use labelme2coco to convert annotations.json, where the annotation is [], what's the matter?

**Comment by udkii** [comment]
I'm labeling for instance segmentation. And I'm curious about the 'group id'. 

For example, I have three classes such as 'car'. 'person', 'traffic light'. 
And there's an image that have a lot of people and cars. And let's say in the picture, a person is standing with the car covered, and the car is divided into two parts. 
Then should I annotate the two parts of the car separately as 'car' and write '0' in the group id? If I write the same number in the group id of both parts, it will be recognized as one car?

And the other (not divided into mutiple parts) cars just annotated with 'car' and do not fill group id?

I'm so confused. Thank you.

**Comment by wkentaro** [comment]
It’s completely up to you and how you parse labelme json file during training.
But in my use case, yes that’s how I meant it being used. 

In my original use case, group id with None (or empty string) value means they are different instances.

**Comment by wkentaro** [comment]
So if you use labelme2voc.py
https://github.com/wkentaro/labelme/blob/master/examples/instance_segmentation/labelme2voc.py

Polygons with the same group id are grouped (recognized) as the same instance.

**Comment by wkentaro** [comment]
But there are other possibilities of using group id to have correspondence between people and objects.
Let's say you have two persons on two bicycles in an image, in this case, maybe you want to annotate which person is riding on which bicycle. You can use `group_id` for this with as the following.
```
{"label": "person", "group_id": "0", shapes:[...]},
{"label": "person", "group_id": "1", shapes:[...]},
{"label": "bicycle", "group_id": "0", shapes:[...]},
{"label": "bicycle", "group_id": "1", shapes:[...]},
```

This use-case is not included in the `examples` folder as there would be various different possibilities on **how to parse and use the JSON file**, and I don't think it is sensible thing to cover all of them in this repo.

**Comment by udkii** [comment]
Ahha I get it. Thank you so much !

**Comment by thegrey0** [comment]
如果我想要标注人物交互的数据集，如何使用group_id,能否显示这个属性呢？
